### ISSUE TITLE
Do not add pages to the sitemap with pagination past the 25,000th item

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/addons-server#17362]
I found this when investigating the the coverage report at https://search.google.com/search-console/index/drilldown?resource_id=https%3A%2F%2Faddons.mozilla.org%2F&item_key=CAMYHyAE&hl=en. In the section about "Submitted URL not found (404)", there are a number of example pages that are category pages for themes with high page numbers.

Here are some examples:
https://addons.mozilla.org/fr/firefox/themes/category/abstract/?page=1696
https://addons.mozilla.org/pl/firefox/themes/category/film-and-tv/?page=1796
https://addons.mozilla.org/pl/firefox/themes/category/nature/?page=1512
https://addons.mozilla.org/de/firefox/themes/category/abstract/?page=2472

Doing a bit of testing, I found that making an API call for any of these with a page number greater than 1000 returns a 404.

I'm not sure if this is expected, but if it is then we shouldn't include any pages above 1000 in the sitemap, and we also shouldn't render pagination links for any page above 1000 in the frontend.

**Comment by diox** [comment]
It is expected. We have configured Elasticsearch in a way that doesn't allow pagination to go past 25000 items (1000 pages of 25). The search API returns a 404 on purpose here.

We shouldn't render pagination links after that, and shouldn't include pages above 1000 in the sitemap.

**Comment by bobsilverberg** [comment]
Thanks @diox. Maybe we can repurpose this issue to fix the sitemap and I will open a separate issue for addons-frontend.

**Comment by AlexandraMoga** [comment]
I've verified the pagination for theme and extension category pages present in the sitemap and it is still correct. I can't verify the 1000 page limit on -dev since we do not have content that reaches 1000 pages. We do have more content on stage though, so I'll be be able to properly verify this issue once the code lands there as well.

**Comment by AlexandraMoga** [comment]
Category page limits are set to 1000 on stage.

This issue is verified fixed on stage.
